aws-lc-rs/src/ed25519.rs

```diff
@@ -269,24 +338,58 @@ impl Ed25519KeyPair {
             return Err(KeyRejected::inconsistent_components());
         }
 
-        let mut derived_public_key = MaybeUninit::<[u8; ED25519_PUBLIC_KEY_LEN]>::uninit();
-        let mut private_key = MaybeUninit::<[u8; ED25519_PRIVATE_KEY_LEN]>::uninit();
-        unsafe {
-            ED25519_keypair_from_seed(
+        let evp_pkey = LcPtr::new(unsafe {
+            EVP_PKEY_new_raw_private_key(EVP_PKEY_ED25519, null_mut(), seed.as_ptr(), seed.len())
+        })?;
+
+        let mut derived_public_key = [0u8; ED25519_PUBLIC_KEY_LEN];
+        let mut out_len: usize = derived_public_key.len();
+        if 1 != unsafe {
+            EVP_PKEY_get_raw_public_key(
+                *evp_pkey.as_const(),
                 derived_public_key.as_mut_ptr().cast(),
-                private_key.as_mut_ptr().cast(),
-                seed.as_ptr(),
-            );
+                &mut out_len,
+            )
+        } {
+            return Err(KeyRejected::unspecified());
+        }
+        debug_assert_eq!(derived_public_key.len(), out_len);
+
+        let mut private_key = [0u8; ED25519_PRIVATE_KEY_LEN];
+        debug_assert_eq!(
+            ED25519_PRIVATE_KEY_LEN,
+            ED25519_PRIVATE_KEY_SEED_LEN + ED25519_PUBLIC_KEY_LEN
+        );
+        let mut out_len: usize = private_key.len();
+        if 1 != unsafe {
+            EVP_PKEY_get_raw_private_key(
+                *evp_pkey.as_const(),
+                private_key.as_mut_ptr(),
+                &mut out_len,
+            )
```

Comment:
Instead of holding on to the "raw bytes" private key in the `Ed25519KeyPair` struct, why not just recompute it on demand?  Maintaining the secret in the struct isn't needed, and it seems unlikely to be needed very often since an `Ed25519KeyPair` can only be constructed from its serialized form.

Reply:
Because `Ed25519KeyPair::seed` awkwardly returns `Seed` which has a lifetime parameter bound to `Ed25519KeyPair` that it uses to reference the private key bytes. If you recall our conversation prior I believe we were in agreement that there was concerns with changing that lifetime parameter?